code/modules/mob/living/living.dm

```diff
@@ -1938,3 +1938,42 @@ Thanks.
 	if(B.host_brain.ckey)
 		to_chat(src, "<span class='danger'>You send a punishing spike of psychic agony lancing into your host's brain.</span>")
 		to_chat(B.host_brain, "<span class='danger'><FONT size=3>Horrific, burning agony lances through you, ripping a soundless scream from your trapped mind!</FONT></span>")
+
+//Returns the thermal mass of the mob's body, excluding reagents but including blood.
+/mob/living/proc/body_thermal_mass()
+
+	//Start with the mass of the body.
+	var/body_thermal_mass = bodymass()
+
+	if(ishuman(src))
```

Comment:
This is a bad way to do this. Never typecheck `src`. Put the behavior you want in the override you want.